lib/server.js

```diff
@@ -1048,38 +1080,20 @@ WalletService.prototype.createTx = function(opts, cb) {
         if (err) return cb(err);
         if (!canCreate) return cb(Errors.TX_CANNOT_CREATE);
 
-        _.each(opts.outputs, function(output) {
-          output.valid = false;
-          var toAddress = {};
-          try {
-            toAddress = new Bitcore.Address(output.toAddress);
-          } catch (ex) {
-            cb(Errors.INVALID_ADDRESS);
-            return false;
-          }
-          if (toAddress.network != wallet.getNetworkName()) {
-            cb(Errors.INCORRECT_ADDRESS_NETWORK);
-            return false;
-          }
-          if (!_.isNumber(output.amount) || _.isNaN(output.amount) || output.amount <= 0) {
-            cb(new ClientError('Invalid amount'));
-            return false;
-          }
-          if (output.amount < Bitcore.Transaction.DUST_AMOUNT) {
-            cb(Errors.DUST_AMOUNT);
-            return false;
-          }
-          output.valid = true;
-        });
-        if (_.any(opts.outputs, {
-          valid: false
-        })) return;
+        if (type != Model.TxProposal.Types.EXTERNAL) {
+		  var validationError = self._validateOutputs(opts, wallet, cb);
+		  if (validationError) {
+			return cb(validationError);
+		  }
+        }
 
         var txOpts = {
           type: type,
           walletId: self.walletId,
           creatorId: self.copayerId,
           outputs: opts.outputs,
+          noOutputsShuffle: true,
```

Comment:
I don't think shuffling should always be disabled...
In fact, maybe we should assume txps of type `EXTERNAL` accept outputs in the order they are received. Would this be a reasonable thing to do?